### PR TITLE
Retrofit feature

### LINKED ISSUE #6: Create a REST client using Retrofit
# REST client with Retrofit

- In a background thread, app queries the `/movie/popular` or `/movie/top_rated` and other API endpoints for the sort criteria for each category
- App requests for related videos for a selected movie via the `/movie/{id}/videos` endpoint in a background thread and display those details when the user selects a movie.
- `/movie/{id}/reviews`
- `movie/now_playing`

`Note: In this phase only create API endpoints. with OkHttp client and the parsing mechanism`

### PR DESCRIPTION
**Pull request by gharras**
added the basic for the call of the TMBD services

Close #6

### CODE REVIEW

File: app/src/main/java/com/ajdi/yassin/netfleak/remote/api/ApiUtils.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    private static final Object sLock = new Object();
+
+    /**
+     * because we are using static method in this class we need to initialize our OkHttpClient
+     * in the static initializer so that we want need to instantiate it each time we do a new
+     * request
+     */
+
+    {
```
Comment (YassinAJDI) [inline app/src/main/java/com/ajdi/yassin/netfleak/remote/api/ApiUtils.java]: missing `static  `key word here

File: .idea/vcs.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="VcsDirectoryMappings">
-    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+    <mapping directory="" vcs="Git" />
```
Comment (YassinAJDI) [inline .idea/vcs.xml]: Why did this change?

File: app/src/main/java/com/ajdi/yassin/netfleak/remote/api/ApiUtils.java (hunk 1 above)
Comment (gharras) [reply app/src/main/java/com/ajdi/yassin/netfleak/remote/api/ApiUtils.java]: either we put static or not is the same for the static block if I remember correctly that why I didn't put it

File: .idea/vcs.xml (hunk 2 above)
Comment (gharras) [reply .idea/vcs.xml]: I didn't change anything on the xml file, it changed alone

File: app/src/main/java/com/ajdi/yassin/netfleak/remote/api/ApiUtils.java (hunk 1 above)
Comment (gharras) [reply app/src/main/java/com/ajdi/yassin/netfleak/remote/api/ApiUtils.java]: I just pushed one with the static keyword

File: .idea/vcs.xml (hunk 2 above)
Comment (YassinAJDI) [inline .idea/vcs.xml]:
```suggestion
    <mapping directory="$PROJECT_DIR$" vcs="Git" />
```

**Review by YassinAJDI — CHANGES_REQUESTED** [review]
LGTM. only one more change and we merge

**Review by YassinAJDI — APPROVED** [review]
LGTM. merging

**Pull request merged**